engine/src/cubos/engine/input/key_with_modifiers.cpp

```diff
@@ -0,0 +1,72 @@
+/// @file
+#include <cubos/core/data/deserializer.hpp>
+#include <cubos/core/data/serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/input/key_with_modifiers.hpp>
+
+using cubos::core::io::Key;
+using cubos::core::io::keyToString;
+using cubos::core::io::Modifiers;
+using cubos::core::io::modifiersToString;
+using cubos::core::io::stringToKey;
+using cubos::core::io::stringToModifiers;
+
+using namespace cubos::engine;
+
+const Key& KeyWithModifiers::key() const
+{
+    return mKey;
+}
+
+const Modifiers& KeyWithModifiers::modifiers() const
+{
+    return mModifiers;
+}
+
+Key& KeyWithModifiers::key()
+{
+    return mKey;
+}
+
+Modifiers& KeyWithModifiers::modifiers()
+{
+    return mModifiers;
+}
+
+bool KeyWithModifiers::pressed() const
+{
+    return mPressed;
+}
+
+void KeyWithModifiers::pressed(bool pressed)
+{
+    if (mPressed == pressed)
+        return;
+
+    mPressed = pressed;
+    CUBOS_DEBUG("Key {} was {}", modifiersToString(mModifiers) + keyToString(mKey), pressed ? "pressed" : "released");
```

Comment:
I know this is useful to debug the Input plugin, but should we keep it from now on? If we do, it should be `_TRACE`